apps/server/src/controllers/dictionaryController.ts

```diff
@@ -133,19 +135,56 @@ export const getSchemaField = async (
 	const dictionary = await dictionaryService.getOneById(dictId);
 	const formattedDictionary = replaceReferences(dictionary);
 
-	const schema = formattedDictionary.schemas.find((schema) => schema.name === schemaName);
+	const schema = formattedDictionary.schemas.find((schema: { name: string }) => schema.name === schemaName);
 
 	if (!schema) {
 		throw new NotFoundError(
 			`Dictionary '${dictionary.name} ${dictionary.version}' does not have a schema named '${schemaName}'`,
 		);
 	}
 
-	const field = schema.fields.find((field) => field.name === fieldName);
+	const field = schema.fields.find((field: { name: string }) => field.name === fieldName);
 	if (!field) {
 		throw new NotFoundError(
 			`Schema '${schemaName}' from Dictionary '${dictionary.name} ${dictionary.version}' does not have a field named '${fieldName}'`,
 		);
 	}
 	res.send(field);
 };
+
+export const downloadTemplates = async (req: Request<{}, {}, {}, { name: string; version: string }>, res: Response) => {
+	const { name, version } = req.query;
+
+	if (!name || !version) {
+		throw new BadRequestError('Missing dictionary name or version in query.');
+	}
+
+	const dictionary = await dictionaryService.downloadDictionaryByNameAndVersion(name, version);
+
+	const zip = new JSZip();
+
+	for (const schema of dictionary.schemas) {
+		const fields = schema.fields || [];
+		const templateRow = fields.reduce(
+			(acc: { [x: string]: string }, field: { name: string | number }) => {
+				acc[field.name] = '';
+				return acc;
+			},
+			{} as Record<string, string>,
+		);
```

Comment:
What you have here is correct and functional, I personally prefer specifying the generic type of the reduce function instead of asserting the type of the initial accumulator value:

```ts
		const templateRow = fields.reduce<Record<string, string>>(
			(acc: { [x: string]: string }, field: { name: string | number }) => {
				acc[field.name] = '';
				return acc;
			},
			{},
		);
```

The advantage here (imo) is to see the type of the accumulator at the start of the function, and then also TS will validate if the initial accumulator value is valid for the accumulator type. This avoids a potential error case where the type assertion declares a type that the value does not match.